### PR TITLE
Update retry logic and add missing test scenarios

### PR DESCRIPTION
**Pull request by jlpadilla**
Story: https://github.com/stolostron/backlog/issues/25059
Bug: https://github.com/stolostron/backlog/issues/25226

### Changes
- Fixes the data validation retry logic and update jest to log when a test retries.
- Add test for search by keyword and search by labels.
- Add authentication test for missing bearer keyword.

### CODE REVIEW

File: tests/api/queries.test.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  afterAll(async () => {
+    execSync(`oc delete ns ${ns}`)
+    if (SEARCH_API_V1) {
+      execSync(`oc delete clusterrole ${usr}`)
+      execSync(`oc delete clusterrolebinding ${usr}`)
+    }
+  }, 15000)
+
+  describe(`using keywords`, () => {
+    test(`should match any resources containing the keyword 'apple'`, async () => {
```
Comment (dislbenn) [inline tests/api/queries.test.js:63]: We need to add the `metadata` for the test cases.

File: tests/api/queries.test.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+  describe(`[${squad}] single request with multiple search queries`, () => {
+    test.todo('should resolve all requests.')
+  })
+
+  describe(`[${squad}] search for relationship`, () => {
+    test.todo('should return relationship count for ${RESOURCE}')
+    test.todo('should return relationship items')
+    test.todo('should return relationship filtered by [releatedKinds]')
+  })
```
Comment (dislbenn) [inline tests/api/queries.test.js:168]: @jlpadilla do we need to include the `$SQUAD` metadata within these `describe` titles? They should receive the metadata from the parent `describe`.

Comment (jlpadilla) [reply tests/api/queries.test.js:168]: Yes, these will get the metadata from the top describe.  I'll remove it when I implement the tests in the next PR.

**Pull request merged**